crates/ruff_python_formatter/src/comments/mod.rs

```diff
@@ -414,6 +414,30 @@ impl<'a> Comments<'a> {
             .parts(&NodeRefEqualityKey::from_ref(node.into()))
     }
 
+    /// Returns any comments on the open parenthesis of a `node`.
+    ///
+    /// For example, `# comment` in:
+    /// ```python
+    /// (  # comment
+    ///    foo.bar
+    /// )
+    /// ```
+    #[inline]
+    pub(crate) fn open_parenthesis_comments<T>(&self, node: T) -> &[SourceComment]
```

Comment:
Should this return an `Option<&SourceComment>` instead, knowing that this can always be at most one comment?

You can use `std::slice::from_ref(comment)` in the formatting code to get a slice from a single element.